fingerprint/src/main/java/com/fingerprint/scanner/HfSecurityFingerprintScanner.kt

```diff
@@ -57,45 +69,18 @@ internal class HfSecurityFingerprint(
 
     override fun disconnect(): Boolean = usbDeviceCommunicator.closeUsbDevice()
 
-    override fun convertImageToBitmapArray(imageData: ByteArray): ByteArray {
-        val bmpHeader: Byte = 14
-        val dibHeader: Byte = 40
-        val headerSize: Byte = (bmpHeader + dibHeader).toByte()
-        val bitmapArray = ByteArray(imageType.size).apply {
-            this[0] = 'B'.code.toByte()
-            this[1] = 'M'.code.toByte()
-            this[10] = headerSize
-            this[11] = 4
-            this[14] = dibHeader
-            this[26] = 1
-            this[28] = 8
-        }
-
-        val imageWidthBytes = intToByteArray(imageType.imageWidth)
-        val imageHeightBytes = intToByteArray(imageType.imageHeight)
-
-        System.arraycopy(imageWidthBytes, 0, bitmapArray, 18, 4)
-        System.arraycopy(imageHeightBytes, 0, bitmapArray, 22, 4)
-
-        for ((j, i) in (headerSize until BMP_DESTINATION_OFFSET step 4).withIndex()) {
-            bitmapArray[i] = j.toByte() // BLUE
-            bitmapArray[i + 1] = j.toByte() // GREEN
-            bitmapArray[i + 2] = j.toByte() // RED
-        }
-
-        // Copying buffer
-        for (i in 0 until (imageType.size - BMP_DESTINATION_OFFSET))
-            bitmapArray[BMP_DESTINATION_OFFSET + i] = imageData[i]
-        return bitmapArray
-    }
-
-    override fun getImageData(): ByteArray? {
+    override suspend fun getImageBytes(): ByteArray? {
         val imageData = ByteArray(imageType.size)
         val result = if (imageType == ScannedImageType.Normal)
             getImageData(imageData)
         else
             getImageDataExtra(imageData)
-        return if (result) imageData else null
+        return if (result) imageData.convertImageDataToBitmapArray(
```

Comment:
This is not related to the PR but you can choose a better name than "result", something that make it appear that it's boolean, like isResultSuccess